src/Rating.php

```diff
@@ -71,42 +82,102 @@ public function getRates($originPostalCode, $postalCode, $weight, array $options
+            'USA.EP' => 'Expedited Parcel USA',
+            'USA.PW.ENV' => 'Priority Worldwide Envelope USA',
+            'USA.PW.PAK' => 'Priority Worldwide pak USA',
+            'USA.PW.PARCEL' => 'Priority Worldwide Parcel USA',
+            'USA.SP.AIR' => 'Small Packet USA Air',
+            'USA.TP' => 'Tracked Packet – USA',
+            'USA.TP.LVM' => 'Tracked Packet – USA (LVM) (large volume mailers)',
+            'USA.XP' => 'Xpresspost USA',
+            'INT.XP' => 'Xpresspost International',
+            'INT.IP.AIR' => 'International Parcel Air',
+            'INT.IP.SURF' => 'International Parcel Surface',
+            'INT.PW.ENV' => 'Priority Worldwide Envelope Int’l',
+            'INT.PW.PAK' => 'Priority Worldwide pak Int’l',
+            'INT.PW.PARCEL' => 'Priority Worldwide parcel Int’l',
+            'INT.SP.AIR' => 'Small Packet International Air',
+            'INT.SP.SURF' => 'Small Packet International Surface',
+            'INT.TP' => 'Tracked Packet – International',
         ];
     }
-    
+
     /**
      * Helper function to extract the option codes.
      *
      * @param array $options
-     *  The options array.
+     *   The options array.
      *
      * @return array
-     *  The list of options with the option-code.
+     *   The list of options with the option-code.
      */
-    protected function parseOptionCodes(array $options) {
-      $valid_options= [];
-      foreach ($options['option_codes'] as $optionCode) {
-        if (!in_array(strtoupper($optionCode), self::getOptionCodes())) {
-          continue;
+    protected function parseOptionCodes(array $options)
+    {
+        $valid_options = [];
+        foreach ($options['option_codes'] as $optionCode) {
+            if (!in_array(strtoupper($optionCode), self::getOptionCodes())) {
+                continue;
+            }
+            // @todo Perhaps we should check for conflicts here, might be overkill.
+            // From Canada Post docs:
+            // There are some options that can be applied to a shipment that
+            // conflict with the presence of another option. You can use the
+            // "Get Option" call in advance to check the contents of the
+            // <conflicting-options> group from a Get Option call for options
+            // selected by end users or options available for a given service.
+            $valid_options[] = [
+                'option-code' => $optionCode,
+            ];
+        }
+
+        return $valid_options;
+    }
+
+    /**
+     * Helper function to extract the service codes.
+     *
+     * @param array $options
+     *   The options array.
+     *
+     * @return array
+     *   The list of services to look up.
+     */
+    protected function parseServiceCodes(array $options)
+    {
+        $services = [];
+        foreach ($options['service_codes'] as $serviceCode) {
+            if (!array_key_exists(strtoupper($serviceCode), self::getServiceCodes())) {
```

Comment:
**CHANGE REQUEST**

Should throw an exception if the caller is requesting a service that does not exist.